### PR TITLE
Add asset family improvements rfc

### PR DESCRIPTION
**Pull request by donaldgray**

### CODE REVIEW

File: docs/rfcs/009-asset-family-improvements.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+The families dictate both how an asset is delivered and how it is processed.
+
+Driven by the requirements for Wellcome's Born Digital work, we have a need to have more control over how assets are delivered and processed - `AssetFamily` is not granular enough.
+
+In summary, `AssetFamily` is now:
+
+* A general indication of what the source file is, this will dictate what values are required (affecting validation rules) and what sort of transcoding/conversion can happen.
+* A preset 'delivery-channel'.
+* A preset image-optimisation policy (`iop`).
+
```
[the remark this answers is not in the document]
Comment (donaldgray) [reply docs/rfcs/009-asset-family-improvements.md:14]: Removed need of AssetFamily, using mediaType instead.

File: docs/rfcs/009-asset-family-improvements.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+## Asset Delivery Channel
+
+To accomodate the need to specify how an asset is delivered, we will introduce a new `delivery-channel` concept which specifies _how_ an asset can be requested and _if_ the engine needs to process. 
+
+Valid values are:
+* `file` - asset will be available on the 'file' path, `/file/{customer}/{space}/{asset}`. This means that the original file is available for download (which could be source mp3, mp4, jpeg etc), possibly in addition to derivatives.
+* `iiif-av` - a timebased derivative of the asset can be streamed on path `/iiif-av/{customer}/{space}/{asset}/{timebased-request}` (e.g. `iiif-av/1/2/foo/full/max/default.mp3`).
```
[the remark this answers is not in the document]
Comment (donaldgray) [reply docs/rfcs/009-asset-family-improvements.md]: The channel doesn't need to be the same as the URL it's called on. I've renamed `iiif-av` to `timebased` so valid values are now `file`, `thumbnails`, `timebased` and `iiif-img`

File: docs/rfcs/009-asset-family-improvements.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+| F            | \*/*      | file             |                  |               | Copy from Origin to storage bucket                          | Stream from storage bucket                                  |
+| F            | \*/*      | iiif-img         | _??_             |               | _Not currently supported but could generate e.g. PDF pages_ | _Could work similar to image handling, proxy to Cantaloupe_ |
+| F            | \*/*      | iiif-av          |                  |               | **invalid**                                                 |                                                             |
+| F            | \*/*      | thumbs           | _??_             | _required_    | _Not currently supported but could generate from key frame_ | _Could work similar to image - predefined thumbs_           |
+
+Notes on the above table:
+* _customer-origin-strategy_ logic will be used for fetching from Origin for `file` delivery channels. In the future we could expand the 'optimised-origin' concept for Engine not to copy and Orchestrator to stream from origin.
```
[the remark this answers is not in the document]
Comment (donaldgray) [reply docs/rfcs/009-asset-family-improvements.md]: As discussed - will open ADR for COS work and do that first.

**Comment by tomcrane** [comment]
I've isolated this commit for what the DDS would do to produce the same behaviour as now, using IOPs and DeliveryChannels rather than AssetFamily.

https://github.com/wellcomecollection/iiif-builder/commit/9e9bcb30a2bed3d82bc3003d948f12e76cb93665

The DDS still has an internal notion of AssetFamily, which is being used here to branch the code. We can retire that as well, but not until after protagonist is bedded in.

Because the DDS has its own Hydra classes, we can support both old and new models on `Image` class:

https://github.com/wellcomecollection/iiif-builder/blob/9e9bcb30a2bed3d82bc3003d948f12e76cb93665/src/Wellcome.Dds/Wellcome.Dds.AssetDomain/Dlcs/Model/Image.cs

**Review by tomcrane — APPROVED** [review]

**Pull request merged**